packages/frontend-2/lib/viewer/composables/ui.ts

```diff
@@ -475,3 +486,66 @@ export function useHighlightedObjectsUtilities() {
     clearHighlightedObjects
   }
 }
+
+export function useViewModeUtilities() {
+  const { instance } = useInjectedViewer()
+
+  // Track the current view mode
+  const currentViewMode = ref<ViewMode>(ViewMode.DEFAULT)
+
+  const setViewMode = (mode: ViewMode) => {
+    const viewModes = instance.getExtension(ViewModes)
+    if (viewModes) {
+      viewModes.setViewMode(mode)
+      currentViewMode.value = mode
+    }
+  }
+
+  return {
+    currentViewMode,
+    setViewMode
+  }
+}
+
+export function useViewerShortcuts() {
+  const { ui } = useInjectedViewerState()
+  const { isSmallerOrEqualSm } = useIsSmallerOrEqualThanBreakpoint()
+
+  const isTypingComment = computed(() => {
+    const isNewThreadEditorOpen = ui.threads.openThread.newThreadEditor.value
+    const isExistingThreadEditorOpen = !!ui.threads.openThread.thread.value
+    return isNewThreadEditorOpen || isExistingThreadEditorOpen
+  })
+
+  const formatKey = (key: string) => {
+    if (key.startsWith('Digit')) {
+      return key.slice(5)
+    }
+    return key
+  }
+
+  const getShortcutDisplayText = (shortcut: ViewerShortcut) => {
+    if (isSmallerOrEqualSm.value) return undefined
+    const modifierText = shortcut.modifiers.join('+')
+    return `${modifierText}+${formatKey(shortcut.key)}`
+  }
+
+  const registerShortcuts = (
+    handlers: Partial<Record<ViewerShortcutAction, () => void>>
+  ) => {
+    Object.values(ViewerShortcuts).forEach((shortcut) => {
+      const handler = handlers[shortcut.action as ViewerShortcutAction]
+      if (handler) {
+        onKeyboardShortcut([...shortcut.modifiers], shortcut.key, () => {
+          if (!isTypingComment.value) handler()
```

Comment:
@andrewwallacespeckle I guess this will not block shortcuts when typing elsewhere (feedback modal, filter input etc?)

Reply:
Good question, will double check and test this. Will leave unresolved until I do

Reply:
I used VueUse here and it seems to work well. You can check it out in this commit:
https://github.com/specklesystems/speckle-server/pull/3651/commits/5a68e558859056a6f791552f3ba0955bae6c9215